File: php7/src/ChainOfResponsibility/Tests/Agent/BranchManagerTest.php
<?php

declare(strict_types = 1);

namespace PetrePatrasc\ChainOfResponsibility\Tests\Agent;

use PetrePatrasc\ChainOfResponsibility\Agent\AbstractLoanAgent;
use PetrePatrasc\ChainOfResponsibility\Agent\BranchManager;

class BranchManagerTest extends AbstractLoanAgentTest
{
    /**
     * @var BranchManager
     */
    protected $branchManager;

    public function setUp()
    {
        parent::setUp();

        $this->branchManager = new BranchManager();
    }

    /**
     * @inheritDoc
     */
    protected function getTestedAgent(): AbstractLoanAgent
    {
        return $this->branchManager;
    }

    /**
     * @inheritDoc
     */
    public function loanAmountDataProvider(): array
    {
        return [
            [100, true],
            [300, true],
            [2000, true],
            [9482.382, true],
            [9999.99, true],
        ];
    }
}

File: java/src/ro/petrepatrasc/chainofresponsability/tests/agent/BranchManagerTest.java
package ro.petrepatrasc.chainofresponsability.tests.agent;

import com.tngtech.java.junit.dataprovider.DataProvider;
import com.tngtech.java.junit.dataprovider.DataProviderRunner;
import org.junit.runner.RunWith;
import ro.petrepatrasc.chainofresponsability.agent.AbstractLoanAgent;
import ro.petrepatrasc.chainofresponsability.agent.BranchManager;

@RunWith(DataProviderRunner.class)
public class BranchManagerTest extends AbstractLoanAgentTest {

    @Override
    public AbstractLoanAgent getTestedAgent() {
        return new BranchManager();
    }

    @DataProvider
    public static Object[][] loanAmountDataProvider() {
        return new Object[][]{
                {371.137, true},
                {8360.28, true},
                {5120.75, true},
                {3985.48, true},
                {9999.99, true},
        };
    }
}
